Update Listener When Props Change

damonbauer opened the pull request:
I ran into #9 and thought I might be able to fix the issue.

I added a `componentWillReceiveProps` lifecycle method that compares incoming & existing props. If the `path` prop is different, it's safe to assume we want to fetch new data. So, I unsubscribed the active listener, set state back to the "default state", and set up a new listener for the new path.

@green-arrow @mikekellyio - if this direction makes sense, I can make the same change to `FirestoreCollection`; I just wanted to make sure this concept made sense before I spent more time on it.

damonbauer commented:
@green-arrow That's a good call. I modified `componentWillReceiveProps` to only set `isLoading: true` state & extracted the `if (this.unsubscribe) { this.unsubscribe() }` logic to be reused.

If you're good with this, I'll take a look at doing the same thing for collections.

green-arrow commented:
@damonbauer - this looks good. Feel free to go ahead and add this to collection.

damonbauer commented:
@green-arrow Have you had a chance to review this?

green-arrow commented on package.json:
```diff
@@ -53,6 +53,7 @@
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
     "kcd-scripts": "^0.32.1",
+    "lodash.isequal": "^4.5.0",
```
@damonbauer - I'd prefer to keep the dependencies on this project to a minimum. Since this is only used in one spot, let's just replace it and remove the dependency.

green-arrow commented on src/FirestoreCollection.js:
```diff
@@ -29,6 +30,20 @@ class FirestoreCollection extends Component {
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!isEqual(nextProps, this.props)) {
```
@damonbauer - Since we're already testing for each individual prop in the tests, we can just do the same here and remove the dependency on `lodash`.

damonbauer commented:
Alright, @green-arrow, I've removed the lodash dependency. I added a small "deep equal" utility that I'm using to compare the `filter` prop, since it can be a nested array. Let me know what you think!
